docs/source/python/api/tables.rst

```diff
@@ -44,6 +44,7 @@ Classes
    RecordBatch
    Table
    TableGroupBy
+   lib.RecordBatchReader
```

Comment:
We shouldn't document things from the `lib` namespace, mostly those are internals.
Also given that the `RecordBatchReader` comes from the ipc implementation module, it should probably get exposed through `pyarrow.ipc` like all the other readers.